src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/MinMax.cs

```diff
@@ -1,21 +1,74 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
 using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Syntax.Nodes;
 using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
 
 namespace Microsoft.PowerFx.Core.Texl.Builtins
 {
     // Min(arg1:n, arg2:n, ..., argN:n)
     // Max(arg1:n, arg2:n, ..., argN:n)
     // Corresponding Excel functions: Min, Max
-    internal sealed class MinMaxFunction : StatisticalFunction
+    internal sealed class MinMaxFunction : BuiltinFunction
     {
         public override bool SupportsParamCoercion => true;
 
+        public override bool IsSelfContained => true;
+
         public MinMaxFunction(bool isMin)
-            : base(isMin ? "Min" : "Max", isMin ? TexlStrings.AboutMin : TexlStrings.AboutMax, FunctionCategories.MathAndStat)
+            : base(isMin ? "Min" : "Max", isMin ? TexlStrings.AboutMin : TexlStrings.AboutMax, FunctionCategories.MathAndStat, DType.Number, 0, 1, int.MaxValue, DType.Number)
+        {
+        }
+
+        public override bool CheckInvocation(TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            var fArgsValid = true;
+            returnType = argTypes[0];
```

Comment:
how does this handle mixed time/date/number?

Reply:
Hmm that's a good question. Should it maybe be an error to compare numbers with datetimes?

Reply:
I've updated my PR to resolve this. Any time there are mixed types, we will coerce to numeric (see final screenshot).